packages/core-backend/src/BackendWebSocketService.test.ts

```diff
@@ -0,0 +1,4837 @@
+import { Messenger } from '@metamask/base-controller';
+
+import {
+  BackendWebSocketService,
+  getCloseReason,
+  WebSocketState,
+  type BackendWebSocketServiceOptions,
+  type BackendWebSocketServiceMessenger,
+  type ClientRequestMessage,
+} from './BackendWebSocketService';
+import { flushPromises } from '../../../tests/helpers';
+
+// =====================================================
+// TYPES
+// =====================================================
+
+// Type for global object with WebSocket mock
+type GlobalWithWebSocket = typeof global & { lastWebSocket: MockWebSocket };
+
+// =====================================================
+// TEST UTILITIES & MOCKS
+// =====================================================
+
+// DOM globals (MessageEvent, CloseEvent, etc.) are now provided by jsdom test environment
+
+/**
+ * Creates a real messenger with registered mock actions for testing
+ * Each call creates a completely independent messenger to ensure test isolation
+ *
+ * @returns Object containing the messenger and mock action functions
+ */
+const createMockMessenger = () => {
+  // Use any types for the root messenger to avoid complex type constraints in tests
+  // Create a unique root messenger for each test
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const rootMessenger = new Messenger<any, any>();
+  const messenger = rootMessenger.getRestricted({
+    name: 'BackendWebSocketService',
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    allowedActions: ['AuthenticationController:getBearerToken'] as any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    allowedEvents: ['AuthenticationController:stateChange'] as any,
+  }) as unknown as BackendWebSocketServiceMessenger;
```

Comment:
Lets clean up the types here to remove 'any' usage.

```suggestion
  const rootMessenger = new Messenger<
    BackendWebSocketServiceAllowedActions,
    BackendWebSocketServiceAllowedEvents
  >();
  const messenger: BackendWebSocketServiceMessenger =
    rootMessenger.getRestricted({
      name: 'BackendWebSocketService',
      allowedActions: ['AuthenticationController:getBearerToken'],
      allowedEvents: ['AuthenticationController:stateChange'],
    });
```